apiary-receiver/apiary-receiver-sqs/src/main/java/com/expedia/apiary/extensions/receiver/sqs/messaging/SqsMessageReader.java

```diff
@@ -0,0 +1,151 @@
+    this.consumer = consumer;
+  }
+
+  @Override
+  public void close() {
+    consumer.shutdown();
+  }
+
+  @Override
+  public Optional<ListenerEvent> read() {
+    if (records == null || !records.hasNext()) {
+      records = receiveMessage();
+    }
+    if (records.hasNext()) {
+      return Optional.of(eventPayLoad(records.next()));
+    } else {
+      return Optional.empty();
+    }
+  }
+
+  public void delete(Message message) {
+    DeleteMessageRequest request = new DeleteMessageRequest()
+        .withQueueUrl(queueUrl)
+        .withReceiptHandle(message.getReceiptHandle());
+    consumer.deleteMessage(request);
+  }
+
+  private Iterator<Message> receiveMessage() {
+      ReceiveMessageRequest request = new ReceiveMessageRequest()
+          .withQueueUrl(queueUrl)
+          .withWaitTimeSeconds(waitTimeSeconds)
+          .withMaxNumberOfMessages(maxMessages);
+      return consumer.receiveMessage(request).getMessages().iterator();
+  }
+
+  private ListenerEvent eventPayLoad(Message message) {
+    try {
+      return messageDeserializer.unmarshal(message.getBody());
+    } catch (Exception e) {
+      throw new SerDeException("Unable to unmarshall event", e);
+    }
+  }
+
+  public static final class Builder {
+    private String queueUrl;
+    private Integer waitTimeSeconds;
+    private Integer maxMessages;
+    private AmazonSQS consumer;
+    private MessageDeserializer messageDeserializer;
+
+    public Builder(String queueUrl) {
+      this.queueUrl = queueUrl;
+    }
+
+    public Builder withConsumer(AmazonSQS consumer) {
+      this.consumer = consumer;
+      return this;
+    }
+
+    public Builder withMessageDeserialiser(MessageDeserializer messageDeserializer) {
+      this.messageDeserializer = messageDeserializer;
+      return this;
+    }
+
+    public Builder withWaitTimeSeconds(Integer waitTimeSeconds) {
+      this.waitTimeSeconds = waitTimeSeconds;
+      return this;
+    }
+
+    public Builder withMaxMessages(Integer maxMessages) {
+      this.maxMessages = maxMessages;
+      return this;
+    }
+
+    public SqsMessageReader build() {
+      checkNotNull(queueUrl);
+
+      consumer = (consumer == null)
+          ? defaultConsumer() : consumer;
+      messageDeserializer = (messageDeserializer == null)
+          ? defaultMessageDeserializer() : messageDeserializer;
+      maxMessages = (maxMessages == null)
+          ? DEFAULT_MAX_MESSAGES : maxMessages;
+      waitTimeSeconds = (waitTimeSeconds == null)
+          ? DEFAULT_POLLING_WAIT_TIME : waitTimeSeconds;
+
+      return new SqsMessageReader(queueUrl, waitTimeSeconds, maxMessages, messageDeserializer, consumer);
+    }
+
+    private AmazonSQS defaultConsumer() {
+      return AmazonSQSClientBuilder.standard().build();
+    }
+
+    private MessageDeserializer defaultMessageDeserializer() {
+      JsonMetaStoreEventDeserializer delegateSerDe = new JsonMetaStoreEventDeserializer(new ObjectMapper());
```

Comment:
The default object mapper should have the property: 

`mapper.configure(FAIL_ON_UNKNOWN_PROPERTIES, false);` set.

Reply:
Refactored this now and added the property in, thanks